xrally_kubernetes/platforms/existing.py

```diff
@@ -12,30 +12,55 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
-from rally.common import cfg
-from rally.common import logging
+import json
+import traceback
+
+from kubernetes import config
+from kubernetes.client.apis import core_v1_api
+from kubernetes.client.apis import version_api
 from rally.env import platform
 
 
-LOG = logging.getLogger(__name__)
+@platform.configure(name="existing", platform="kubernetes")
+class KubernetesPlatform(platform.Platform):
+    """Default plugin for Kubernetes."""
 
-CONF = cfg.CONF
+    CONFIG_SCHEMA = {
+        "type": "object",
+        "properties": {
+            "config_file": {
+                "type": "string"
+            }
+        },
+        "additionalProperties": False
+    }
 
+    def __init__(self, spec, **kwargs):
+        super(KubernetesPlatform, self).__init__(spec, **kwargs)
+        self._api = None
 
-@platform.configure(name="existing", platform="kubernetes")
-class Kubernetes(platform.Platform):
-    """Default plugin for Kubernetes platform
-
-    It may be used to test any existing Kubernetes API compatible cluster.
-    """
     def create(self):
-        # TBD
-        return {}, {}
+        """Pass kubeconfig to internal Configuration."""
+        return {"config_file": self.spec.get("config_file")}, {}
 
     def destroy(self):
-        # NOTE(boris-42): No action need to be performed.
+        # NOTE(prazumovsky): No action need to be performed.
         pass
 
+    def check_health(self):
+        """Check whatever platform is alive."""
+        try:
+            api = config.new_client_from_config(self.spec.get("config_file"))
```

Comment:
As for the spec described above (in my comment with jsonschema) I used the next code for initialization of k8sclient:```


        from kubernetes import client as k8s_config
        from kubernetes.client import api_client
        from kubernetes.client.apis import core_v1_api

        # NOTE(andreykurilin): KubernetesClient doesn't provide any __version__
        #   property to identify the client version (you are welcome to fix
        #   this code if I'm wrong). Let's check for some backward incompatible
        #   changes to identify the way to communicate with it.
        if hasattr(k8s_config, "ConfigurationObject"):
            # Actually, it is `k8sclient < 4.0.0`, so it can be
            #   kubernetesclient 2.0 or even less, but it doesn't make any
            #   difference for us
            self._k8s_client_version = 3
        else:
            self._k8s_client_version = 4

        if self._k8s_client_version == 3:
            config = k8s_config.ConfigurationObject()
        else:
            config = k8s_config.Configuration()

        config.host = self._spec["server"]
        config.ssl_ca_cert = self._spec["certificate-authority"]
        config.cert_file = self._spec["client-certificate"]
        config.key_file = self._spec["client-key"]
        if self._spec.get("tls_insecure", False):
            config.verify_ssl = False

        if self._k8s_client_version == 3:
            api = api_client.ApiClient(config=config)
        else:
            api = api_client.ApiClient(config)

        client = core_v1_api.CoreV1Api(api)
```